### PR TITLE
WIP: implement proxy support

### LINKED ISSUE #23: HTTP_PROXY
Is there a way to specify the proxy? We're behind a coorporate web proxy, and the agent doesn't seem to use the HTTP_PROXY or HTTPS_PROXY environment variables.

### PR DESCRIPTION
**Pull request by virth**
should resolve #23

### CODE REVIEW

**Comment by virth** [comment]
My implementation seems to actually connect to the specified Server (in local tests) but is somehow still not working on our server with our coorperate proxy-server. I just don't know if my implementation is wrong or if the tunnel-mechanism in `api-client.getAuthToken` is just something that our proxy-server is not supporting.

The error: `Auth error:  RequestError tunneling socket could not be established, cause=getaddrinfo ENOTFOUND`

This seems to be HTTP-404 but when I try call the proxy-server with a browser I actually get a response.

Could somebody have a look at my changes and take a wild guess?

File: lib/api-client.js [hunk 1]
```diff
@@ -23,7 +24,39 @@ module.exports.getAuthToken = function(config, agentName, socket) {
...
     debug(url);
     debug(postdata);
-    return got.post(url, { body: postdata, json: true, headers: { 'user-agent': config.UA } })
+
+    var proxy = process.env.http_proxy;
+    var agent;
+    if (proxy) {
+        log('Using proxy server: ' + proxy);
+        if (url.startsWith('https') && proxy.startsWith('https')) {
+            agent = tunnel.httpsOverHttps({
```
Comment (BorisWilhelms) [inline lib/api-client.js]: Options object can be created once and passed to the methods.

File: lib/api-client.js [hunk 2]
```diff
@@ -23,7 +24,39 @@ module.exports.getAuthToken = function(config, agentName, socket) {
...
+                proxy: {
+                    host: proxy
```
Comment (BorisWilhelms) [inline lib/api-client.js]: The value passed the host property must only be the host part, without protocol and port.

**Comment by phagunbaya** [comment]
What is the plan for merging this PR ?

**Comment by virth** [comment]
@phagunbaya I hope I can finish this in the next two weeks

**Pull request closed without merging**